### ISSUE TITLE
JVM target error

### ISSUE THREAD

**Issue by avjiang** [korlibs/klock#76]
Hi, i am facing problem after updated IntelliJ.  I also have updated my kotlin version to 1.3.60
  
    IntelliJ IDEA 2019.2.1 (Community Edition)
    Build #IC-192.6262.58, built on August 20, 2019
    Runtime version: 11.0.3+12-b304.39 x86_64
    VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

 
Previous version i never face any issue, but in this build, i face this error. Now I am using version 1.8.0

    Cannot inline bytecode built with JVM target 1.8 into bytecode that is being built with JVM target 1.6. Please specify proper '-jvm-target' option
FAILURE: Build failed with an exception.

Here is the code in my commonMain folder
    
    var timeZone = TimeSpan((8.hours).milliseconds)

I have changed my project JVM version setting to 1.6. 
![image](https://user-images.githubusercontent.com/19270457/69615468-e190d100-106f-11ea-8d9c-592922ef8531.png)

Is there anything that I have to do to solve this issue? Hopefully you can guide me on this. Thanks!

**Comment by soywiz** [comment]
I started to use Target JVM version 1.8

**Comment by avjiang** [comment]
Is there any thing I need to adjust in my side?

**Comment by soywiz** [comment]
https://kotlinlang.org/docs/reference/using-gradle.html#attributes-specific-for-jvm

```
jvmTarget = "1.8"
```

If you are using multiplatform:
https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html#configuring-compilations

**Comment by avjiang** [comment]
Hi, i am still facing the issue. Please enlighten me. Below is my settings

    apply plugin: 'kotlin-multiplatform'

    kotlin {
    jvm {
        compilations.main.kotlinOptions {
            // Setup the Kotlin compiler options for the 'main' compilation:
            jvmTarget = "1.8"
        }

        compilations.main.compileKotlinTask // get the Kotlin task 'compileKotlinJvm'
        compilations.main.output // get the main compilation output
        compilations.test.runtimeDependencyFiles // get the test runtime classpath
    }

    // Configure all compilations of all targets:
    targets.all {
        compilations.all {
            kotlinOptions {
                allWarningsAsErrors = true
            }
        }
    }

    sourceSets {
        commonMain {
            dependencies {
                implementation "com.soywiz.korlibs.klock:klock:$klockVersion" // 1.8.0
            }
        }
    }

kotlin version is 1.3.60

**Comment by avjiang** [comment]
Hi @soywiz, is there any update about this issue? 

I have tried to use the 1.8.0 library version with kotlin version = 1.3.60, but still encounter JVM target issue. I did not face any problem below version 1.8.0. 

Please help... Thanks

**Comment by soywiz** [comment]
You have to set the `kotlinOptions.jvmTarget` on your build.gradle file to "1.8".

Like here:
https://github.com/korlibs/easy-kotlin-mpp-gradle-plugin/blob/7373efde446827e965d92c6a61bf8d00637e22bc/src/main/kotlin/com/soywiz/korlibs/targets/JVM.kt#L16

**Comment by avjiang** [comment]
I have included the jvm target to my build.gradle. The sample code is on the previous message. Is it wrong?
-------- Original message --------From: Carlos Ballesteros Velasco <notifications@github.com> Date: 07/01/2020  6:50 pm  (GMT+08:00) To: korlibs/klock <klock@noreply.github.com> Cc: avjiang <vinfoster0107@gmail.com>, Author <author@noreply.github.com> Subject: Re: [korlibs/klock] JVM target error (#76) You have to set the kotlinOptions.jvmTarget on your build.gradle file to "1.8".
Like here:
https://github.com/korlibs/easy-kotlin-mpp-gradle-plugin/blob/7373efde446827e965d92c6a61bf8d00637e22bc/src/main/kotlin/com/soywiz/korlibs/targets/JVM.kt#L16

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/korlibs/klock/issues/76?email_source=notifications\u0026email_token=AETAWOJJXGEIDR3NA5Z4JO3Q4RM63A5CNFSM4JRUFB42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIIPH6A#issuecomment-571536376",
"url": "https://github.com/korlibs/klock/issues/76?email_source=notifications\u0026email_token=AETAWOJJXGEIDR3NA5Z4JO3Q4RM63A5CNFSM4JRUFB42YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIIPH6A#issuecomment-571536376",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by soywiz** [comment]
Can you generate a small project reproducing the issue in a repo or a zip file? It should work. But if it doesn't I investigate it

**Comment by avjiang** [comment]
Noted. I will provide u a sample application soon. Ty in advance!

**Comment by avjiang** [comment]
hi @soywiz , I Have attached a sample app. Thanks!

[sample_app.zip](https://github.com/korlibs/klock/files/4033566/sample_app.zip)

**Comment by soywiz** [comment]
Sorry for the huge delay. I didn't know your problem was with the android. The `jvm` config is for jvm desktop. You have to specify kotlinOptions.jvmTarget for android too:

Change:

```
jvm("android")
```

with

```
jvm("android") {
    compilations.getting {
        kotlinOptions {
            // Setup the Kotlin compiler options for the 'main' compilation:
            jvmTarget = "1.8"
        }
    }
}
```

**Comment by avjiang** [comment]
hi @soywiz , 

where is the jvm("android") located at? Sorry if i asked too many questions. I am new to this kotlin multiplatform. 

Thanks!

**Comment by avjiang** [comment]
hi @soywiz 

Since our current project is using groovy, i tried to find proper syntax to it and i come out with below settings

    jvm("android") {
            compilations.main.kotlinOptions {
                jvmTarget = "1.8"
            }
        }

and seems like it doesn't throw the jvm target error anymore. I tried to get you sample app by using groovy, but the tutorial is using kotlin, not groovy anymore. 

So i guess it has been fixed now. Thank you so much for your help!
